ARCHITECTURE.md

```diff
@@ -0,0 +1,123 @@
+# Qiskit.org architecture overview
+
+Technical overview of how the code is structured and how it all comes together.
+
+## Table of contents
+
+- [Vue.js components](#vuejs-components)
+- [Vuex store](#vuex-store)
+- [Design system](#design-system)
+- [CSS](#css)
+- [Markdown content](#markdown-content)
+- [Testing](#testing)
+- [Third-party integrations](#third-party-integrations)
+
+## Vue.js components
+
+We write our components as [single file components](https://vuejs.org/v2/guide/single-file-components.html).
+
+The **script section** is written in [class-style syntax](https://class-component.vuejs.org/) with TypeScript and uses decorators provided by [**vue-property-decorator**](https://github.com/kaorun343/vue-property-decorator).
+
+The **style section** is written in SCSS.
+
+Here is an example of a component called `GreetingMessage.vue`:
```

Comment:
From https://github.com/Qiskit/qiskit.org/pull/1225#discussion_r560228814 by @delapuente:

> Would be convenient to provide an example based on a real one.

My goal with this code snippet is to give a very quick glance at the use of TypeScript and SCSS in our Vue components. I believe that using a real example may add more elements and divert to attention to other more complex parts of a component.

Nevertheless I see value in displaying a code snipped with an example of a simplified version of a real project component.